gcloud/_helpers.py

```diff
@@ -159,6 +165,41 @@ def _app_engine_id():
     return app_identity.get_application_id()
 
 
+def _file_project_id():
+    """Gets the project id from the credentials file if one is available.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from JSON credentials file if value exists,
+              else ``None``.
+    """
+    credentials_file_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS')
+    if credentials_file_path:
+        with open(credentials_file_path, 'r') as credentials_file:
+            credentials_data = credentials_file.read()
+            credentials = json.loads(credentials_data)
+            return credentials.get('project_id')
```

Comment:
Malformed as in they possibly misnamed the `project_id` property? Otherwise if it's bad JSON it will throw a parsing error.

If `project_id` isn't found then it will just move on to the next method of detecting the project ID.

I think it would be good to throw something here though.
What exception should we throw here if the `project_id` property is missing?